src/db/parity_db.rs

```diff
@@ -189,12 +305,248 @@ impl DBStatistics for ParityDb {
+
+                        if let Some(e) = error_iterating {
+                            Err(e.into())
+                        } else {
+                            Ok::<_, anyhow::Error>(())
+                        }
+                    });
+                }
+                DbColumn::GraphFull => {
+                    tasks.spawn(async move {
+                        let mut iter = db.iter(col as u8)?;
+                        while let Ok(Some(entry)) = iter.next() {
+                            let (cid, block) = (Cid::try_from(entry.0)?, entry.1);
+                            tx_clone.send_async((cid, block)).await?;
+                            progress_inc();
+                        }
+                        Ok::<_, anyhow::Error>(())
+                    });
+                }
+                DbColumn::Other => {
+                    panic!("Other column is not exportable");
+                }
+            }
+            debug!("Finished iterating the {col} column");
+        }
+
+        drop(tx);
+
+        while let Some(res) = tasks.join_next().await {
+            let _out = res?;
+        }
+
+        let mut writer = writer.lock().await;
+        writer.flush().await?;
+        writer.close().await?;
+
+        Ok(())
+    }
+}
+
 #[cfg(test)]
 mod test {
+    use fvm_ipld_car::CarReader;
     use parity_db::CompressionType;
+    use tokio_util::compat::{TokioAsyncReadCompatExt, TokioAsyncWriteCompatExt};
+
+    use crate::{
+        blocks::BlockHeader, db::tests::db_utils::parity::TempParityDB, shim::address::Address,
+    };
 
     use super::*;
 
+    #[test]
+    fn total_exportable_entries_with_stats_test() -> anyhow::Result<()> {
+        let mut db = TempParityDB::new();
+        let data = b"h'nglui mglw'nafh Cthulhu R'lyeh wgah'nagl fhtagn";
+
+        assert_eq!(0, db.total_exportable_entries()?);
+
+        // write to the simplified column, this should be counted
+        db.put_keyed(&Cid::new_v1(DAG_CBOR, Blake2b256.digest(data)), data)?;
+
+        db.force_flush();
+        assert_eq!(1, db.total_exportable_entries()?);
+
+        // write to the full column, this should be counted
+        db.put_keyed(
+            &Cid::new_v1(DAG_CBOR, cid::multihash::Code::Sha2_256.digest(data)),
+            data,
+        )?;
+        db.force_flush();
+        assert_eq!(2, db.total_exportable_entries()?);
+
+        // write to the other column, this should NOT be counted
+        db.write("dagon", "bloop")?;
+        db.force_flush();
+        assert_eq!(2, db.total_exportable_entries()?);
+
+        Ok(())
+    }
+
+    // This test will:
+    // * write important entries to the database,
+    // * dump the database to a CAR file,
+    // * read the CAR file and verify that the entries are present
+    #[tokio::test]
+    async fn write_exportable_test() -> anyhow::Result<()> {
+        let mut db = TempParityDB::new();
+        let data = [
+            b"h'nglui mglw'nafh Cthulhu R'lyeh".to_vec(),
```

Comment:
I would probably invest some time looking into dumping a bigger database and then trying to read the car file / validate it comparing to what `export` does.

Just to see if there are any performance issues related to the ordering.